onnxruntime/python/_pybind_state.py

```diff
@@ -10,4 +10,5 @@
 try:
     from onnxruntime.capi.onnxruntime_pybind11_state import *  # noqa
 except ImportError as e:
-    warnings.warn("Cannot load onnxruntime.capi. Error: '{0}'".format(str(e)))
+    warnings.warn("Cannot load onnxruntime.capi. Error: '{0}'.".format(str(e)) +
```

Comment:
I don't think we can assume all exceptions caught here are related to missing VS C++ runtime. It can happen with any missing dependent shared libs. (on Linux even)

If the VC runtime dll is always installed in well-known locations, can we check for it in 
https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/python/onnxruntime_validation.py ?

Reply:
That is true. We can't assume all exceptions caught here are related to the missing VC runtime. I am not sure we can add a robust enough check to specifically point out that the target machine is missing the VC runtime. Also we have to be careful about inserting checks in onnxruntime_validation because someone might be building from source using VS 2017 and we don't want to be checking for the 2019 runtime when ORT doesn't need it. As for the missing shared libs on Linux causing this, maybe I can amend the message to include - "on Windows" ?

Reply:
maybe check if we're running on windows before printing a windows specific error message.

Reply:
Sure. I ll push another iteration.

On a slightly related note, do you know where onnxruntime_validation is called from ? It seems to have this logic - 
```
        if int(__my_distro_ver__.split('.')[0]) < 11:
            warnings.warn('Unsupported macOS version (%s). ONNX Runtime supports macOS 11.0 or later.' %
                          (__my_distro_ver__))
```
I am not sure if this logic is correct given that all modern Mac OS versions re currently 10.x.

Reply:
Checking for a specific library and version is an implementation detail of the new compiler, something that might change. Hopefully it works always. @RyanUnderhill is there a better way to detect if vs2019 redist is installed on a machine?

Reply:
AFAIK - it is very hard to precisely determine if the 2019 VC runtime has been installed on the target machine in Python. VS 2015, VS 2017, VS 2019 all share the same redist files. For example, VS 2015, VS 2017 and VS 2019 all come with vcruntime140.dll. A user who had installed VS 2017 would have the vcruntime140.dll (with a different version number stamped) on their machine - so we would have to try and parse the version number from the dll and try and ascertain which runtime is installed. As a proxy to this - I went for checking if the vcruntime140_1.dll has been installed (since we know for sure that the 2019 VC runtime is th only one that ships with this dll). I have a comment included that says - "As a proxy...." 

Starting the next VS version, I guess they should have vcruntime140_1.dll in their newer runtimes otherwise it would break the promised binary compatibility - https://docs.microsoft.com/en-us/cpp/porting/binary-compat-2015-2017?view=vs-2019. So t that time, if we are using the next VS to build, we have to look for a different way to print out a better error message (as a person who has VS 2019 might have vcruntime140_1.dll but we actually require the next version of the runtime)